app/try_ruby.rb

```diff
@@ -61,7 +62,9 @@ def initialize
 
     #If hold down the control and the Enter key goes down, run
     $document.on :keydown, '#editor' do |e|
-      if e.key == "Enter" && e.ctrl?
+      if e.key == "Enter" && ((e.ctrl? && !@navigator.user_agent&.match?(/\b(iPad|iPhone|iPod)\b/)) ||
```

Comment:
This is kinda hard to read... maybe we could add helper functions like:

```ruby
def ios?
  @navigator.user_agent&.match?(/\b(iPad|iPhone|iPod)\b/)
end

def macos?
  @navigator.user_agent&.match?(/\bMac\b/)
end
```

Then the modified code could be just:

```ruby
    $document.on :keydown, '#editor' do |e|
      if e.key == "Enter" && ((!ios? && e.ctrl?) || ((ios? || macos?) && e.meta?))
        e.prevent
        do_run
      end
    end
```

Reply:
I added the Helper class to make my code more reusable